core/types/block_ext_test.go

```diff
@@ -93,13 +92,17 @@ func exportedFieldsDeepCopied[T interface {
 }
 
 // assertDifferentPointers asserts that `a` and `b` are both non-nil
-// pointers of the same type and point to different memory locations.
+// pointers pointing to different memory locations.
 func assertDifferentPointers[T any](t *testing.T, a *T, b any) {
 	t.Helper()
-	require.NotNilf(t, a, "a (%T) cannot be nil", a)
-	require.NotNilf(t, b, "b (%T) cannot be nil", b)
-	require.IsType(t, a, b, "test is badly written: a and b must be of the same type")
-	if a == b {
+	switch {
+	case a == nil:
+		t.Errorf("a (%T) cannot be nil", a)
+	case b == nil:
+		t.Errorf("b (%T) cannot be nil", b)
+	case a == b:
 		t.Errorf("pointers to same memory")
 	}
+	// Note: no need to check `b` is of the same type as `a`, otherwise
+	// the memory address would be different as well.
```

Comment:
(fyi) Not technically true, but sufficient for these purposes so no need to change it.

Reply:
If `a` is not nil so it has a memory address, can there be another-type variable `b` at that same memory address? 🤔  How? Why?

Reply:
Memory can be interpreted however one wishes; e.g. this [valid use](https://pkg.go.dev/unsafe#Pointer:~:text=(1)%20Conversion%20of%20a%20*T1%20to%20Pointer%20to%20*T2.) of `unsafe.Pointer`. The chance of that happening here is negligible, hence why I said "sufficient for these purposes", but the reason I originally put a check is to be explicit instead of relying on assumptions.